src/main/java/com/twilio/http/NetworkHttpClient.java

```diff
@@ -30,17 +30,17 @@ public class NetworkHttpClient extends HttpClient {
     /**
      * Create a new HTTP Client.
      */
-    public NetworkHttpClient() {
-        this(DEFAULT_REQUEST_CONFIG);
+    public NetworkHttpClient(final List<String>... userAgentExtensions) {
+        this(DEFAULT_REQUEST_CONFIG, userAgentExtensions);
```

Comment:
Again, not in favor of the additional overloading, but understand the issues with building the headers as part of the constructor. Is there a path where the extensions are set/added _after_ construction and then included in when calling `makeRequest`?

Reply:
addressed, within makeRequest we can add header
(I have tested that as well by creating a dummy service and calling that service by adding header similar to makeRequest.)